src/views/RecIM/components/List/index.jsx

```diff
@@ -38,13 +39,37 @@ const ParticipantList = ({
   teamID,
   showInactive,
   callbackFunction,
+  adminPage,
 }) => {
   if (!participants?.length)
     return <Typography className={styles.secondaryText}>No participants to show.</Typography>;
   let content = participants.map((participant) => {
     if (!showInactive && participant.Role === 'Inactive') {
       return null;
     }
+    if (adminPage) {
+      return (
+        <ParticipantListing
+          key={participant.username}
+          participant={participant}
+          minimal={minimal}
+          withAttendance={withAttendance}
+          adminPage={true}
+          initialAttendance={
+            withAttendance && attendance?.find((att) => att.Username === participant.Username)
+          }
+          isAdmin={isAdmin}
+          matchID={matchID}
+          teamID={teamID}
+          callbackFunction={(bool) => callbackFunction(bool)}
+          showParticipantOptions={
+            showParticipantOptions &&
+            participant.Role !== 'Team-captain/Creator' &&
+            participant.Role !== 'Requested Join' // don't promote people who haven't joined
+          }
+        />
+      );
+    }
     return (
       <ParticipantListing
         key={participant.username}
```

Comment:
```suggestion
      <ParticipantListing
        key={participant.username}
        adminPage={adminPage}
```
you can actually shrink your code down a bit here by doing this:

What this does is it will pass `adminPage` from your `ParticipantList` parameter to your listings. That way, you don't need the `if` statement and you get to cut down on some repeated code!